### PR TITLE
169650165- FT users search for accommodations

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?
Enables users to search for accommodations on the barefoot nomad system. 

#### Description of Task to be completed?
Allow users to search for accommodation by Name, Description & location. 

Any of the three can be used to search.

#### How should this be manually tested?

1. Clone this Repo.
2. Run ` npm install `
3. Run ` sequelize db:migrate `
4. Run  ` sequelize db:seed:all `
5. Run npm run dev
6. Using Postman send a  ` GET ` request to ` /api/v1/accommodations/search `
7. Add Query Parameters to the GET request. i.e. 
```
location
name
description
```
#### Any background context you want to provide?

N/ A

#### Screenshots (if appropriate)
Screen Shot 2019-11-12 at 11 14 26
Screen Shot 2019-11-12 at 11 15 13
Screen Shot 2019-11-12 at 11 15 28
Screen Shot 2019-11-12 at 11 15 43
Screen Shot 2019-11-12 at 11 15 54

#### Questions:

N/A

### CODE REVIEW

**Review by alainmateso — CHANGES_REQUESTED** [review]
Impressive work @cheza. However, it would be much more nice if the user could search by amenities and highlights as well. Also, please make sure a relevant message is returned when no accommodations are found to match the provided search condition!

File: src/middlewares/verifySearchKeys.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      || key === 'highlights'
+      || key === 'amenities') {
+      return true;
+    }
+    return false;
+  });
+
+  const invalidKeys = searchKeys.filter(key => key === false);
+  if (invalidKeys.length > 0) {
+    return responseHelper(res, 'You Provided Invalid Search Keys. Only Name, Description & Locaton Allowed.', null, 400);
```
Comment (nignanthomas) [inline src/middlewares/verifySearchKeys.js]: Kindly update the `validation error message` to also include `amenities` and `highlights`.

**Review by nignanthomas — CHANGES_REQUESTED** [review]
Awesome job @Cheza-Dzabala 
However, I left a comment.

**Review by alainmateso — APPROVED** [review]

**Review by nignanthomas — APPROVED** [review]

File: src/controllers/searchController.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import accommodationServices from '../services/accommodationServices';
+import responseHelper from '../utils/responseHelper';
+import searchQuery from '../utils/db/queries/accommodationSearchQuery';
+
+export default class SearchController {
+  static accommodations = async (req, res) => {
+    const { query } = req;
+    try {
+      const accommodations = await accommodationServices.findAccommodations(searchQuery(query));
```
Comment (nakiwuge) [inline src/controllers/searchController.js:12]: Add a new line

File: src/middlewares/verifySearchKeys.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import responseHelper from '../utils/responseHelper';
+
+export default (req, res, next) => {
+  const { query } = req;
+  const keys = Object.keys(query);
```
Comment (nakiwuge) [inline src/middlewares/verifySearchKeys.js:6]: nakiwuge now Member
Add a new line

**Pull request merged**